change-validation/src/main/scala/com/normation/plugins/changevalidation/WorkflowService.scala

```diff
@@ -99,22 +102,39 @@ class EitherWorkflowService(cond: () => Box[Boolean], whenTrue: WorkflowService,
 
 object TwoValidationStepsWorkflowServiceImpl {
   case object Validation extends WorkflowNode {
-    val id = WorkflowNodeId("Pending validation")
+    val id: WorkflowNodeId = WorkflowNodeId("Pending validation")
   }
 
   case object Deployment extends WorkflowNode {
-    val id = WorkflowNodeId("Pending deployment")
+    val id: WorkflowNodeId = WorkflowNodeId("Pending deployment")
   }
 
   case object Deployed extends WorkflowNode {
-    val id = WorkflowNodeId("Deployed")
+    val id: WorkflowNodeId = WorkflowNodeId("Deployed")
   }
 
   case object Cancelled extends WorkflowNode {
-    val id = WorkflowNodeId("Cancelled")
+    val id: WorkflowNodeId = WorkflowNodeId("Cancelled")
   }
 
   val steps: List[WorkflowNode] = List(Validation, Deployment, Deployed, Cancelled)
+
+  /*
+   * Validation rule logic:
+   * - check for self validation (only needed if current user is the CR author)
+   * - check for current user rights.
+   *   WARNING: WE ARE SIDE STEPPING authz check until https://issues.rudder.io/issues/22595 is solved
+   */
+  private def canValidate(isCreator: Boolean, selfValidation: () => Box[Boolean]): Boolean = {
+    val correctActor = selfValidation().getOrElse(false) || !isCreator
+    correctActor && CurrentUser.checkRights(AuthorizationType.Validator.Edit)
+  }
+
```

Comment:
These two methods are the relevant changes: we use `CurrentUser.checkRights` in place of a comparison on string. That methods correctly check for the `any` right that admin has.